src/braket/experimental/autoqasm/errors.py

```diff
@@ -24,12 +22,28 @@ class UnknownQubitCountError(AutoQasmError):
     """Missing declaration for the number of qubits."""
 
     def __init__(self):
-        self.message = f"""Unspecified number of qubits.
+        self.message = """Unspecified number of qubits.
+
+Specify the number of qubits used by your program by supplying the \
+`num_qubits` argument to `autoqasm.function`. For example:
 
-Please declare the total number of qubits for your program. \
-You can do that by calling your AutoQASM program with the \
-{ProgramOptions.NUM_QUBITS.value} keyword argument. \
+    @autoqasm.function(num_qubits=5)
+    def my_autoqasm_function():
+        ...
 """
 
     def __str__(self):
         return self.message
+
+
+class InconsistentNumQubits(AutoQasmError):
+    """Num qubits supplied to main function does not match subroutine."""
+
+    def __init__(self):
+        self.message = """\
+The number of qubits specified by one of your subroutines does not match the \
+arguments supplied to your main function. Remove the num_qubits decorator \
+argument to all aq.functions except the one you wish to call."""
```

Comment:
above you use `autoqasm.function`, and here you call it `aq.function`. I'm not sure which we should prefer (I lean toward `aq.function` since this is what all of our examples will use) but we should at least be consistent.

Reply:
True. I also prefer `aq.function`, but technically there is no `aq`. The error message with `aq` would not be meaningful to someone using `import autoqasm` instead of `import autoqasm as aq`. I would say documentation should use the full name, and tutorials / examples should use `aq` (and anything that has the opportunity to include import statements)

Reply:
I'll update for consistency